app/views/pages/get-started/build-your-first-app/building-the-user-interface.liquid

```diff
@@ -0,0 +1,195 @@
+---
+metadata:
+  title: Building the user interface
+  description: Follow these step-by-step tutorials to build a simple To do application on platformOS.
+converter: markdown
+---
+
+
+The very basic building block of any web application is it’s HTML structure. platformOS uses Liquid as a language for creating HTML templates.
+
+In this context, you can think of _Liquid_ as an addition to your HTML that will allow you to split code into smaller fragments, render some repeating parts without actually repeating the code or showing the data pulled from the database on the page.
+
+Liquid is parsed and executed **server-side**. All of the operations using Liquid are computed first on the server, and then the result of those operations is send to user browser.
+
+
+
+## Build the main application layout
+
+The main page’s HTML structure on platformOS is kept in `app/views/layouts/application.html.liquid`. Notice the file extension is `.liquid` – this is going to be a pattern for most of the files related to page’s structure. Specifically, the file named `application.html.liquid` is automatically used as a default structure for all pages in your application.
+
+Let’s start with a very simple HTML structure:
+
+#### app/views/layouts/application.html.liquid
+
+{% raw %}
+```liquid
+<!DOCTYPE html>
+<html>
+	<head>
+		<meta charset="utf-8">
+		<title>To do list</title>
+	</head>
+	<body>
+		<h1>My to do list</h1>
+
+		{{ content_for_layout }}
+	</body>
+</html>
+```
+{% endraw %}
+
+You may notice a special {% raw %}`{{ content_for_layout }}`{% endraw %} tag. This little thing here is Liquid.
+
+There will be many other tags and uses of Liquid in the future. This particular one is later automatically replaced by the actual page content that we will define in other `.liquid` files. So the content of other pages will be injected in place for {% raw %}`{{ content_for_layout }}`{% endraw %}. This makes the layout file a universal page structure for all of our content.
+
+{% include 'alert/tip', content: 'Remember that you need to [synchronise](/get-started/working-with-the-code-and-files/#upload-code-changes-automatically-when-you-save-the-file) or deploy your changes to the instance to see the results.' %}
+
+We can see this in action by creating another file – this one with the content that will be shown on the index page of our application.
+
+All of the pages (or _subpages_) of our web application needs to be placed in a separate folder: `app/views/pages/`. Start by creating the index file with a simple text:
+
+#### app/views/pages/index.liquid
+
+```html
+<h2>Hello world!</h2>
+```
+
+Now, when you go to your instance URL, you should see the application title that we defined in `application.html.liquid` as well as the content of `index.liquid`. Check out the source, it’s all there!
```

Comment:
probably we should be consistent with format - either we add it everywhere or we don't, i.e. we use application.html.liquid and index.html.liquid or we use application.liquid and  index.liquid . I'd suggest we use .html.liquid

Reply:
As discussed, removed `.html` for now, and it would be nice to extend the documentation with an article on "How to build a REST API for your application" that would introduce the concept of file types on pOS.